### ISSUE TITLE
map.fitBounds(bounds) differs from map.getBounds() output.

### ISSUE THREAD

**Issue by j** [maplibre/maplibre-gl-js#1778]
We're trying to save state of a map via user input and have come across an issue where when we save `map.getBounds` and re-create it, `map.getBounds` changes:

```typescript
    const map = new mapLibre.Map({
        container,
        bounds,
        style: getMapStyleSpecification(container, lineWidthScale),
        maxZoom: 14,
        minZoom: 2,
        preserveDrawingBuffer: true,
        attributionControl: false
    });

    map.fitBounds(bounds); // map.getBounds() does not match `bounds` given from above even though it came from previous `map.getBounds()` call
```

**Comment by HarelM** [comment]
Fit bounds can behave differently depending on the screen resolution (or more correctly the screen aspect ratio).
Get bounds get the corners of the screen.
So fit bounds and get bounds are not symmetrical operations.

**Comment by j** [comment]
@HarelM yes, I understand and figured that. But how could I enforce exact bounds when aspect ratio does not change?

**Comment by HarelM** [comment]
How do you decide which bounds to set? I guess you can use some projection from screen pixels to lat lng...

**Comment by HarelM** [comment]
I'm not sure this is an issue though. Feels more like a discussion.
Can I convert it to a discussion?

**Comment by j** [comment]
I suppose it can be a discussion depending on how you look at it.

I feel that setting a map's bounds, getting those bounds, saving them in a DB and initializing that same map with those bounds that creates different bounds in the same exact container with set aspect ratio can potentially be an issue.  (Sorry for the run on sentence).  

We attempted react state and kept getting re-renders because props didn't match but I couldn't spend too much time on it.
